DigitalLearningSolutions.Web/Controllers/LearningPortalController/Current.cs

```diff
@@ -170,6 +170,37 @@ public async Task<IActionResult> LaunchLearningResource(int learningLogItemId)
             return Redirect(learningResourceLink);
         }
 
+        [HttpGet]
+        [SetDlsSubApplication(nameof(DlsSubApplication.LearningPortal))]
+        [ServiceFilter(typeof(VerifyDelegateCanAccessActionPlanResource))]
+        [Route("/LearningPortal/Current/ActionPlan/{learningLogItemId:int}/MarkAsComplete")]
+        public async Task<IActionResult> MarkActionPlanResourceAsComplete(int learningLogItemId)
+        {
+            var actionPlanResource = await actionPlanService.GetActionPlanResource(learningLogItemId);
+            var model = new MarkActionPlanResourceAsCompleteViewModel(learningLogItemId, actionPlanResource!.Name);
+            return View("Current/MarkActionPlanResourceAsComplete", model);
+        }
+
+        [HttpPost]
+        [ServiceFilter(typeof(VerifyDelegateCanAccessActionPlanResource))]
+        [Route("/LearningPortal/Current/ActionPlan/{learningLogItemId:int}/MarkAsComplete")]
+        public IActionResult MarkActionPlanResourceAsComplete(

```

Comment:
Sorry, just realised that this should probably have the "SetDlsSubApplication" as well in case of data validation error.

Reply:
Ah right, added it here